### PR TITLE
Multisite Role Support Added

### PR DESCRIPTION
**Pull request by marslanabdulrauf**
**Description:**
This PR adds Multi site Role Support

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-5409

### CODE REVIEW

File: openedx/core/djangoapps/user_authn/views/login.py [hunk 1]
```diff
@@ -429,9 +429,10 @@ def login_user(request):
         if possibly_authenticated_user is None or not possibly_authenticated_user.is_active:
             _handle_failed_authentication(user, possibly_authenticated_user)
 
+        # TODO: handle is_blocked from our custom model, is_active ko nh handle krna.
```
Comment (taimoor-ahmed-1) [inline openedx/core/djangoapps/user_authn/views/login.py:432]: has this been implemented?

File: openedx/features/edly/migrations/0012_auto_20230418_1310.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    user=django_user
+                )
+                edly_access_user.groups.add(*groups)
+                edly_access_user.is_blocked = user.is_blocked
+                edly_access_user.save()
+
+
+def backward_migration(apps, schema_editor):
+    EdlyMultiSiteAccess = apps.get_model('edly', 'EdlyMultiSiteAccess')
+    EdlyMultiSiteAccess.objects.all().delete()
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/migrations/0012_auto_20230418_1310.py:30]: will the `backward_migration` run each time?

File: openedx/features/edly/tests/test_middleware.py [hunk 3]
```diff
@@ -128,38 +138,38 @@ def test_user_without_edly_organization_access_and_with_frontend_logout_url(self
...
+    #     Test logged in staff user has access to all sites.
+    #     """
+    #     edly_user = UserFactory(is_staff=True)
+    #     client = Client()
+    #     client.login(username=edly_user.username, password='test')
 
-        response = client.get('/', follow=True)
-        assert response.status_code == 200
+    #     response = client.get('/', follow=True)
+    #     assert response.status_code == 200
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/tests/test_middleware.py:172]: kindly uncomment test or remove it if duplicate

File: openedx/features/edly/tests/test_utils.py [hunk 4]
```diff
@@ -86,15 +85,12 @@ def setUp(self):
...
-        self.user = UserFactory.create()
-        self.admin_user = UserFactory.create(is_staff=True)
         self.user_password = 'password'
         self.request = RequestFactory().get('/')
-        self.request.user = self.user
         self.request.session = self._get_stub_session()
         self.request.site = SiteFactory()
 
-        self.edly_user_profile = EdlyUserProfileFactory(user=self.user)
+        # self.edly_user_profile = EdlyUserProfileFactory(user=self.user)
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/tests/test_utils.py:93]: kindly remove this comment

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

File: openedx/features/edly/migrations/0012_auto_20230418_1310.py (hunk 2 above)
Comment (marslanabdulrauf) [reply openedx/features/edly/migrations/0012_auto_20230418_1310.py:30]: nope, it only runs when we we explicitly run backward migration
`python manage.py migrate app_name 0000`

**Review by taimoor-ahmed-1 — APPROVED** [review]

**Pull request merged**